cedar-policy-core/src/ast/id.rs

```diff
@@ -274,7 +276,7 @@ impl AnyId {
     ///
     /// This method was created for the `From<cst::Ident> for AnyId` impl to use.
     /// See notes on `Id::new_unchecked()`.
-    pub(crate) fn new_unchecked(s: impl Into<SmolStr>) -> AnyId {
+    pub fn new_unchecked(s: impl Into<SmolStr>) -> AnyId {
```

Comment:
Just curious why we need this to be `pub` instead of using `.parse()` as suggested in the comment.  Should we update the comment?

Reply:
Reverted. Use `from_str` (i.e., `.parse()`) instead.